notifications/src/main/kotlin/notifications/init/NotificationInitialization.kt

```diff
@@ -0,0 +1,100 @@
+package notifications.init
+
+import com.github.kittinunf.result.failure
+import com.github.kittinunf.result.map
+import io.reactivex.schedulers.Schedulers
+import iroha.protocol.Commands
+import mu.KLogging
+import notifications.service.NotificationService
+import notifications.service.TransferNotifyEvent
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import sidechain.iroha.IrohaChainListener
+import sidechain.iroha.NOTARY_DOMAIN
+import sidechain.iroha.util.getTransferCommands
+import java.math.BigDecimal
+import java.util.concurrent.Executors
+
+/**
+ * Notifications initialization service
+ */
+@Component
+class NotificationInitialization(
+    @Autowired private val irohaChainListener: IrohaChainListener,
+    @Autowired private val notificationService: NotificationService
+) {
+
+    /**
+     * Initiates notification service
+     * @param onIrohaChainFailure - function that will be called in case of Iroha failure
+     */
+    fun init(onIrohaChainFailure: () -> Unit) {
+        irohaChainListener.getBlockObservable().map { irohaObservable ->
+            irohaObservable
+                .subscribeOn(Schedulers.from(Executors.newSingleThreadExecutor()))
+                .subscribe({ block ->
+                    //Get transfer commands from block
+                    getTransferCommands(block).forEach { command ->
+                        val transferAsset = command.transferAsset
+                        // Notify deposit
+                        if (isDeposit(transferAsset)) {
+                            handleDepositNotification(transferAsset)
+                        }
+                        // Notify withdrawal
+                        else if (isWithdrawal(transferAsset)) {
+                            handleWithdrawalEventNotification(transferAsset)
+                        }
+                    }
+                }, { ex ->
+                    logger.error("Error on Iroha subscribe", ex)
+                    onIrohaChainFailure()
+                })
+        }
+    }
+
+    // Checks if deposit event
+    private fun isDeposit(transferAsset: Commands.TransferAsset): Boolean {
+        return transferAsset.srcAccountId.endsWith("@$NOTARY_DOMAIN")
```

Comment:
Do we need to include `@`? Are you assuming we are going to have a domain like `@blablablanotary`? In that case it is better to assign `@notary` to `NOTARY_DOMAIN` unless it is used somewhere else separately

Reply:
Yes, I think that `@blablanotarty` may happen. `NOTARY_DOMAIN` represents the domain. `notary` is a valid domain, while `@notary` is not.